sherpa/astro/fake.py

```diff
@@ -21,111 +21,158 @@
+
+    If simulated backgrounds are required then they should be
+    simulated with this routine and then added to the faked source
+    dataset.
+
+    This routine was significantly changed in Sherpa 4.16.0. To update
+    old code, note that the following arguments are no-longer used:
+    is_source, pileup_model, add_bkgs, bkg_models and id. The model
+    argument must now include any background model terms, and the
+    necessary scaling term for the conversion from the background to
+    source aperture (a combination of the add_bkgs and
+    bkg_models="model" arguments), and also include the response
+    necessary to convert to channels and counts (a combination of the
+    is_source and pileup_model arguments). Setting the bkg_models
+    argument to a PHA dataset is now handled by adding the background
+    to the data first, and then setting include_bkg_data to True.
 
     Examples
     --------
-        Estimate the signal from a 5000 second observation using the
-        ARF and RMF from "src.arf" and "src.rmf" respectively:
-
-        >>> set_source(1, xsphabs.gal * xsapec.clus)
-        >>> gal.nh = 0.12
-        >>> clus.kt, clus.abundanc = 4.5, 0.3
-        >>> clus.redshift = 0.187
-        >>> clus.norm = 1.2e-3
-        >>> fake_pha(1, 'src.arf', 'src.rmf', 5000)
-
-        Simulate a 1 mega second observation for the data and model
-        from the default data set. The simulated data will include an
-        estimated background component based on scaling the existing
-        background observations for the source. The simulated data
-        set, which has the same grouping as the default set, for
-        easier comparison, is created with the 'sim' label and then
-        written out to the file 'sim.pi':
-
-        >>> arf = get_arf()
-        >>> rmf = get_rmf()
-        >>> bkg = get_bkg()
-        >>> bscal = get_backscal()
-        >>> grp = get_grouping()
-        >>> qual = get_quality()
-        >>> texp = 1e6
-        >>> set_source('sim', get_source())
-        >>> fake_pha('sim', arf, rmf, texp, backscal=bscal, bkg=bkg,
-        ...          grouping=grp, quality=qual, grouped=True)
-        >>> save_pha('sim', 'sim.pi')
+
+    Simulate the data from an absorbed APEC model, where the response
+    is manually created (a "perfect" RMF and the ARF is flat, with a
+    break at 3.2 keV), and then use the Sherpa plot objects to display
+    the simulated data and model. Note that the exposure time must be
+    set (if the model normalization is set).
+
+    >>> import numpy as np
+    >>> from sherpa.astro.data import DataPHA
+    >>> from sherpa.astro.instrument import Response1D, create_arf, create_delta_rmf
+    >>> from sherpa.astro.xspec import XSphabs, XSapec
+    >>> gal = XSphabs()
+    >>> clus = XSapec()
+    >>> gal.nh = 0.12
+    >>> clus.kt = 4.5
+    >>> clus.abundanc = 0.3
+    >>> clus.redshift = 0.23
+    >>> clus.norm = 6.3e-4
+    >>> chans = np.arange(1, 101, dtype=np.int16)
```

Comment:
This example is so long, some intermediate explanation might help.
```suggestion
    >>> # Generate ARF and RMF. In this example, we generate them from scratch, but in pratice
    >>> # they are typically read in from a file
    >>> chans = np.arange(1, 101, dtype=np.int16)
```

Reply:
It's one of the issues I have with the "documentation can also be a test" approach, as sometimes it leads to a lot of extra code...

Reply:
Even if it's not a test, it's very useful to a beginner that they can copy and paste the example and it "just works". Sometimes if can't be done, but the risk in those artificially short examples that don't do any of the setup needed is always that it's just a repeat of the function signature and not a working example that someone can take to modify for their own situation. So, I'm absolutely in favor of long examples - even if they are not used as tests.
(In the past, in Sherpa and other packages, I've sometimes looked at the test code to find an actual complete example, but test code is not written for that. It's often littered with pytest parmeterization and fixtures that do not help the beginner.